qcodes/instrument_drivers/Keysight/N52xx.py

```diff
@@ -0,0 +1,436 @@
+                           get_cmd='SENS:AVER:COUN?',
+                           get_parser=int,
+                           set_cmd='SENS:AVER:COUN {:d}',
+                           unit='',
+                           vals=Numbers(min_value=1,max_value=65536))
+
+        # Setting frequency range
+        self.add_parameter('start',
+                           label='Start Frequency',
+                           get_cmd='SENS:FREQ:STAR?',
+                           get_parser=float,
+                           set_cmd='SENS:FREQ:STAR {}',
+                           unit='',
+                           vals=Numbers(min_value=min_freq,max_value=max_freq))
+        self.add_parameter('stop',
+                           label='Stop Frequency',
+                           get_cmd='SENS:FREQ:STOP?',
+                           get_parser=float,
+                           set_cmd='SENS:FREQ:STOP {}',
+                           unit='',
+                           vals=Numbers(min_value=min_freq,max_value=max_freq))
+
+        # Number of points in a sweep
+        self.add_parameter('points',
+                           label='Points',
+                           get_cmd='SENS:SWE:POIN?',
+                           get_parser=int,
+                           set_cmd='SENS:SWE:POIN {}',
+                           unit='',
+                           vals=Numbers(min_value=1, max_value=100001))
+
+        # Electrical delay
+        self.add_parameter('electrical_delay',
+                           label='Electrical Delay',
+                           get_cmd='CALC:CORR:EDEL:TIME?',
+                           get_parser=float,
+                           set_cmd='CALC:CORR:EDEL:TIME {:.6e}',
+                           unit='s',
+                           vals=Numbers(min_value=0, max_value=100000))
+
+        # Sweep Time
+        self.add_parameter('sweep_time',
+                           label='Time',
+                           get_cmd='SENS:SWE:TIME?',
+                           get_parser=float,
+                           unit='s',
+                           vals=Numbers(0,1e6))
+        # Sweep Mode
+        self.add_parameter('sweep_mode',
+                           label='Mode',
+                           get_cmd='SENS:SWE:MODE?',
+                           set_cmd='SENS:SWE:MODE {}',
+                           vals=Enum("HOLD", "CONT", "GRO", "SING"))
+
+        # Traces
+        # Note: These will be accessed through the traces property which updates
+        # the channellist to include only active trace numbers
+        self._traces = ChannelList(self, "PNATraces", PNATrace)
+        self.add_submodule("traces", self._traces)
+        # Add shortcuts to trace 1
+        trace1 = PNATrace(self, "tr1", 1)
+        for param in trace1.parameters.values():
+            self.parameters[param.name] = param
+        # By default we should also pull any following values from this trace
+        self.write("CALC:PAR:MNUM 1")
```

Comment:
similar command is used 2 times in the `PNATrace` class.
- could you extract it as a separate method of `PNATrace` class?
- could you re-use the extracted method here since you've created the `trace1` object?

Reply:
To address your concern, I wrapped this into a parameter, `active_trace`, and am writing this instead.